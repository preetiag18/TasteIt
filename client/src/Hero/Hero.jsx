import React from "react";
import { NavLink } from "react-router-dom";

import RiceVideo from "./Rice.mp4";
import classnames from "classnames";

import classes from "./Hero.module.css";

const Hero = () => {
  return (
    <div className={classes.container}>
      <video className={classes.bgVideo} autoPlay loop muted>
        <source src={RiceVideo} type="video/mp4"></source>
      </video>
      <div className={classes.caption}>
        <h1 className={classnames(classes.heading, classes.large_font)}>
          TasteIt
        </h1>
        <h2 className={classnames(classes.heading, classes.medium_font)}>
          TasteIt is the recipe app which is made in React22S group lessons
        </h2>
        <NavLink to="recipes" className={classes.heading}>
          <button className={classes.button}>Browse recipes</button>
        </NavLink>
      </div>
    </div>
  );
};

export default Hero;
