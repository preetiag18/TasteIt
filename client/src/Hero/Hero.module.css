.container {
  position: relative;
}

.bgVideo {
  object-fit: cover;
  width: 100%;
  height: 25rem;
}

.caption {
  position: absolute;
  top: 2rem;
  left: 2rem;
}

.heading {
  margin-bottom: 2rem;
  color: white;
}
.large_font {
  font-size: 700%;
}
.medium_font {
  font-size: 200%;
}

.button {
  background-color: #111827;
  border: 1px solid transparent;
  border-radius: 0.75rem;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  flex: 0 0 auto;
  font-family: "Inter var", ui-sans-serif, system-ui, -apple-system, system-ui,
    "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
    "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-size: 1.125rem;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.75rem 1.2rem;
  text-align: center;
  text-decoration: none #6b7280 solid;
  text-decoration-thickness: auto;
  transition-duration: 0.2s;
  transition-property: background-color, border-color, color, fill, stroke;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  width: auto;
}

.button:hover {
  background-color: #374151;
}

.button:focus {
  box-shadow: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
}

@media (min-width: 768px) {
  .button {
    padding: 0.75rem 1.5rem;
  }
}

@media (max-width: 768px) {
  .large_font {
    font-size: 500%;
  }
}
