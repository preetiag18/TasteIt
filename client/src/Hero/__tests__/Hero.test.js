import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Hero from "../Hero";
import { BrowserRouter } from "react-router-dom";

describe("Hero", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Hero />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
