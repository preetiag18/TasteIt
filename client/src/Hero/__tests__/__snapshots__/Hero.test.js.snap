// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div
  className="container"
>
  <video
    autoPlay={true}
    className="bgVideo"
    loop={true}
    muted={true}
  >
    <source
      src="Rice.mp4"
      type="video/mp4"
    />
  </video>
  <div
    className="caption"
  >
    <h1
      className="heading large_font"
    >
      TasteIt
    </h1>
    <h2
      className="heading medium_font"
    >
      TasteIt is the recipe app which is made in React22S group lessons
    </h2>
    <a
      className="heading"
      href="/recipes"
      onClick={[Function]}
    >
      <button
        className="button"
      >
        Browse recipes
      </button>
    </a>
  </div>
</div>
`;
